### ISSUE TITLE
Compiler Error CS0250 – destructor is outdated term

### ISSUE THREAD

**Issue by bravequickcleverfibreyarn** [dotnet/docs#26068]
Regard [Finalizers (C# Programming Guide)](https://docs.microsoft.com/en-us/dotnet/csharp/programming-guide/classes-and-structs/destructors) is _destructor_ former naming.

> Finalizers (historically referred to as destructors) are used to perform any necessary final clean-up when a class instance is being collected by the garbage collector.

**Compiler Error CS0250**
> Do not directly call your base class Finalize method. It is called automatically from your destructor.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 4c96f063-d987-bfda-60d9-8b9ff4f444e5
* Version Independent ID: 604f15c1-0202-765e-445a-1d783b9b89b6
* Content: [Compiler Error CS0250](https://docs.microsoft.com/en-us/dotnet/csharp/misc/cs0250?f1url=%3FappId%3Droslyn%26k%3Dk(CS0250))
* Content Source: [docs/csharp/misc/cs0250.md](https://github.com/dotnet/docs/blob/main/docs/csharp/misc/cs0250.md)
* Product: **dotnet-csharp**
* Technology: **csharp-diagnostics**
* GitHub Login: @BillWagner
* Microsoft Alias: **wiwagn**

**Comment by BillWagner** [comment]
Thanks for noting this @Uzivatel919

I've added this to our backlog to address when we next update this article.

I've also added the "up-for-grabs" label on this and similar issues. There are likely many occurrences of destructor when the current preferred term is finalizer. The ECMA committee made decided to prefer the term "finalizer" over "destructor" during the C# 5.0 edits. That version was formally adopted in December 2017.

We'd appreciate any PRs that make updates to these terms.
